python/triton/language/core.py

```diff
@@ -643,15 +667,12 @@ def __eq__(self, other: block_type) -> bool:
             return False
         return self.element_ty == other.element_ty and self.shape == other.shape
 
-    def __ne__(self, other: block_type) -> bool:
-        return not self.__eq__(other)
-
     @property
     def scalar(self):
         return self.element_ty
 
 
-class tuple_type(dtype):
+class tuple_type(base_type):
```

Comment:
`dtype` was being treated like a type base class before, but I chose to split it so that only tensor related types are dtype subclasses.